OtripleS.Portal.Web.Tests.Unit/Services/Courses/CourseServiceTests.Exceptions.RetrieveAll.cs

```diff
@@ -0,0 +1,130 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Moq;
+using OtripleS.Portal.Web.Models.Courses;
+using OtripleS.Portal.Web.Models.Courses.Exceptions;
+using Xunit;
+
+namespace OtripleS.Portal.Web.Tests.Unit.Services.Courses
+{
+    public partial class CourseServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(CriticalApiExceptions))]
+        public async Task ShouldThrowCriticalDependencyOnRetrieveAllIfCriticialErrorOccursAndLogItAsync(
+            Exception criticalDependencyException)
+        {
+            // given
+            var failedCourseDependencyException =
+                new FailedCourseDependencyException(
+                    criticalDependencyException);
+
+            var expectedCourseDependencyException =
+                new CourseDependencyException(
+                    failedCourseDependencyException);
+
+            this.apiBrokerMock.Setup(broker =>
+                broker.GetAllCoursesAsync())
+                    .ThrowsAsync(criticalDependencyException);
+
+            // when
+            ValueTask<List<Course>> retrievedCoursesTask =
```

Comment:
I would argue that we should call this `retrieveAllCoursesTask`
Retrieved insinuates that courses are already here but they are to be retrieved.

Reply:
If you agree, please do that on other places too.